servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -104,7 +104,7 @@ abstract class Http4sServlet[F[_]](service: HttpApp[F], servletIo: ServletIo[F])
       headers = toHeaders(req),
       body = servletIo.reader(req),
       attributes = Vault.empty
-        .insert(Request.Keys.PathInfoCaret, getPathInfoIndex(req, uri))
+        .insert(Request.Keys.PathInfoCaret, getPathInfoIndex(req, uri).get)
```

Comment:
I would say it falls in the same category as the `get` a few lines below. That is: it will throw if the `Servlet` implementation is wrong or our implementation is wrong.

What I did here (and what I usually do in such situations) is that I'd rather have my code fail, than fall back to an incorrect behaviour. Have we had this `get` instead of a `getOrElse` here, we would have never released this bug.

Other thing is that it (as well as the other 4 gets, 3 asInstanceOfs and 2 unsafeFromString) would throw inside a `Either#map` (that's what the `yield` clause will translate to). Maybe to prevent that we could wrap the whole thing in `org.http4s.ParseResult#fromTryCatchNonFatal` and flatMap it.

Reply:
Hi @danicheg 
I addressed these concerns in the latest commit (https://github.com/http4s/http4s/pull/5793/commits/c928a3e0a27e6a9bff27653f6c5c15763d794335)